### DISCUSSION TITLE
There are two situations how to solve?

### DISCUSSION THREAD

**Discussion by hello-soon** [jonataslaw/getx#1619] [Ideas]
hi, There are two situations how to solve?

Some pages do not require authentication, such as normalPage1, normalPage2, ..., normalPageN
Some pages require authentication, such as authPage1, authPage2, ..., authPageN
"LoginPage" for logging in for authentication

One situation is this:
1,  normalPage1  goto authPage
2, How GetX monitors this route? In the callback, we can check the authentication.  If there is no auth, we will go to 'LoginPage' instead of 'authPage'.
3, In the "LoginPage", after a successful login, how can automatically jump to the "authPage"

Another situation is this:
1 now in LoginPage
2 goto authPage2 ( call by system notification),   but now there is no auth, will goto another LoginPage, how to prevent？

**Comment by pckimlong** [comment]
You can use GetX "Worker" and place it inside OnInit method.
